src/pvf.rs

```diff
@@ -1,15 +1,38 @@
 use anyhow::anyhow;
 use futures::channel::oneshot;
-use polkadot_node_core_pvf::{Config, PvfPrepData, ValidationHost, PrepareJobKind};
+use polkadot_node_core_pvf::{Config, PrepareJobKind, PvfPrepData, ValidationHost};
 use polkadot_parachain::primitives::ValidationCode;
 use std::path::PathBuf;
+use std::process::Command;
 use std::time::{Duration, Instant};
 
-pub async fn setup_pvf_worker(pvfs_path: PathBuf) -> ValidationHost {
-    let program_path = std::env::current_exe().expect("current_exe failed?");
+pub async fn setup_pvf_worker(pvfs_path: PathBuf) -> anyhow::Result<ValidationHost> {
```

Comment:
Yeah the code duplication here is a bit unfortunate. It would be nice to split out the logic on my side into separate `get_prepare_worker_paths` and `get_execute_worker_paths` functions, and you could just use the first one. It would be a nice refactor but it's not as simple as it sounds, will punt for now and do it later if there's time.

Reply:
I think it's fine. This tool will include the preparation worker binary instead of using one installed on the system to make it easier to do stuff in CI for release preparation.